### PR TITLE
[DO NOT REVIEW ][DO NOT MERGE] sint_heap_undeclared resolved

### PR DESCRIPTION
**Pull request by yaswanth278**

### CODE REVIEW

File: os/tools/check_package_size.py [hunk 1]
```diff
@@ -113,8 +113,9 @@ def check_binary_header(bin_type):
 COMMON_PART_TMP_SIZE = FLASH_SIZE
 APP1_PART_TMP_SIZE = FLASH_SIZE
 APP2_PART_TMP_SIZE = FLASH_SIZE
-
+size=len(SIZE_LIST)
```
Comment (amogh-samsung) [inline os/tools/check_package_size.py:116]: Requesting you to follow certain coding guidelines: - 
There should always be a space before and after operators

File: os/tools/check_package_size.py [hunk 2]
```diff
@@ -113,8 +113,9 @@ def check_binary_header(bin_type):
...
 for name in NAME_LIST :
+    if PART_IDX>size : break
```
Comment (amogh-samsung) [inline os/tools/check_package_size.py:118]: Same as above, coding guidelines

File: apps/shell/tash_main.c [hunk 3]
```diff
@@ -121,12 +121,12 @@ char *tash_read_input_line(int fd)
 	int nbytes;
 	int char_idx;
 	bool is_tab_pressed = false;
-#if TASH_MAX_STORE > 0
+	int prev_cmd_len = 0;
```
Comment (amogh-samsung) [inline apps/shell/tash_main.c:124]: Please use tab-spaces at the beginning of the line of code instead of multiple single spaces

File: os/tools/check_package_size.py [hunk 4]
```diff
@@ -113,8 +113,9 @@ def check_binary_header(bin_type):
...
     PART_IDX += 1
     if name == "kernel" or name == "os" :
         if KERNEL_PART_TMP_SIZE > (int(SIZE_LIST[PART_IDX]) * 1024) :
```
Comment (amogh-samsung) [inline os/tools/check_package_size.py:121]: Request you to mention the issue https://github.com/Samsung/TizenRT/issues/5845 for which this fix is intended in the PR description

**Review by amogh-samsung — CHANGES_REQUESTED** [review]

**Comment by amogh-samsung** [comment]
Hello @yaswanth278 , could you merge these 2 commits as well since they seem to be resolving one common issue only and perhaps give the commit a stronger/more proper title please?

File: os/arch/arm/src/common/up_internal.h [hunk 5]
```diff
@@ -278,7 +278,7 @@ EXTERN uint32_t _edata;			/* End+1 of .data */
 EXTERN uint32_t _sbss;			/* Start of .bss */
 EXTERN uint32_t _ebss;			/* End+1 of .bss */
 EXTERN uint32_t _sidle_stack;		/* Start of idle stack */
-EXTERN uint32_t __sint_heap_start;	/* Start of heap in internal RAM region */
+EXTERN uint32_t _sint_heap_start;	/* Start of heap in internal RAM region */
```
Comment (abhishek-samsung) [inline os/arch/arm/src/common/up_internal.h:281]: Other than this change other file changes seem to be not related to the issue. Could you please check and revert those related changes?

File: os/arch/arm/src/common/up_internal.h [hunk 6]
```diff
@@ -278,7 +278,7 @@ EXTERN uint32_t _edata;			/* End+1 of .data */
...
 EXTERN uint32_t __sext_heap_start;	/* Start of heap in external RAM region */
```
Comment (abhishek-samsung) [inline os/arch/arm/src/common/up_internal.h:282]: @thapav , 
in the previous commit (https://github.com/Samsung/TizenRT/commit/b36bd18a57093fbaa6c23aa0d3c26621c119d5e6) where _sint_heap_start and _sext_heap_start were added, in up_internal.h they were added with an extra underscore. Could you please confirm if its requried?

Comment (thapav) [reply os/arch/arm/src/common/up_internal.h:282]: > @thapav , in the previous commit ([b36bd18](https://github.com/Samsung/TizenRT/commit/b36bd18a57093fbaa6c23aa0d3c26621c119d5e6)) where _sint_heap_start and _sext_heap_start were added, in up_internal.h they were added with an extra underscore. Could you please confirm if its requried?

It must be '__s**_heap_start'
Please refer to https://github.com/Samsung/TizenRT/blob/master/docs/HowToAddnewBoard.md

**Review by ewoodev — COMMENTED** [review]
Could you please add commit detail comment?

https://github.com/Samsung/TizenRT/wiki/Documentations#commit-rules

File: build/configs/rtl8730e/loadable_apps/defconfig [hunk 7]
```diff
@@ -0,0 +1,1520 @@
+#
+# Automatically generated file; DO NOT EDIT.
+# TinyAra Configuration
+#
+
+#
+# Build Setup
+#
+# CONFIG_EXPERIMENTAL is not set
```
Comment (amogh-samsung) [inline build/configs/rtl8730e/loadable_apps/defconfig:9]: Hi, it seems that some commits were merged while your PR was still open for review.

You need to fetch the latest code and re-apply your commits/changes on top of it and push again for consistency.

File: lib/libc/machine/arm/Make.defs [hunk 8]
```diff
@@ -62,4 +62,6 @@ else ifeq ($(CONFIG_ARCH_CORTEXM7),y)   # Cortex-M7 is ARMv7E-M
 include ${TOPDIR}/../lib/libc/machine/arm/armv7-m/Make.defs
 else ifeq ($(CONFIG_ARCH_ARMV8M_FAMILY),y)   # Cortex-M33/M55 is ARMv8-M
 include ${TOPDIR}/../lib/libc/machine/arm/armv7-m/Make.defs
+else ifeq ($(CONFIG_ARCH_ARMV7A_FAMILY),y)   # ARMV7A
+include ${TOPDIR}/../lib/libc/machine/arm/armv7-a/Make.defs
```
Comment (amogh-samsung) [inline lib/libc/machine/arm/Make.defs:66]: For Eg- this file does not relate to your issue
This change is probably there because other commits were merged in the course of your PR review.

File: lib/libc/machine/arm/armv7-a/Make.defs [hunk 9]
```diff
@@ -0,0 +1,45 @@
+###########################################################################
+#
+# Copyright 2023 Samsung Electronics All Rights Reserved.
```
Comment (amogh-samsung) [inline lib/libc/machine/arm/armv7-a/Make.defs:3]: Not a part of your changes

**Review by amogh-samsung — CHANGES_REQUESTED** [review]
Please re-base and re-apply your changes
Please go through the changes in the PR after you re-apply to confirm that there are only those changes which were made by you and related to the issue at hand.

File: os/arch/arm/src/stm32l4/stm32l4_ospi.c [hunk 10]
```diff
@@ -1920,9 +1920,7 @@ FAR struct mtd_dev_s *up_flashinitialize(void)
     priv->bwrite = this_bwrite;
     priv->read = this_read;
     priv->ioctl = this_ioctl;
-#ifdef CONFIG_MTD_REGISTRATION
-    priv->mtd.name = "stm32l4_flash";
-#endif
+
```
Comment (amogh-samsung) [inline os/arch/arm/src/stm32l4/stm32l4_ospi.c:1923]: Why is a single extra-line addition change made here?
I think this might be due to some re-base confusion.
Is this related to the issue?

File: os/tools/check_package_size.py (hunk 1 above)
Comment (amogh-samsung) [inline os/tools/check_package_size.py:116]: this does not seem related to the issue that we are trying to fix here

File: build/configs/rtl8730e/Make.defs [hunk 11]
```diff
@@ -121,6 +121,9 @@ ifeq ($(CONFIG_FRAME_POINTER),y)
 endif
 
 ARCHCFLAGS += -fno-builtin 
+ifeq ($(CONFIG_APP_BINARY_SEPARATION),y)
+  ARCHCFLAGS += -fno-common
+endif
```
Comment (amogh-samsung) [inline build/configs/rtl8730e/Make.defs]: Unrelated changes, kindly check

File: apps/shell/tash_main.c [hunk 12]
```diff
@@ -121,12 +121,12 @@ char *tash_read_input_line(int fd)
...
 	bool is_esc_pressed = false;
+	char direction;
+#if TASH_MAX_STORE > 0
```
Comment (amogh-samsung) [inline apps/shell/tash_main.c:127]: Please add a description to the commit (1-2 lines) explaining the idea behind making this change to solve the issue at hand.

**Comment by amogh-samsung** [comment]
This PR has many changes not-relating to the issue of the undeclared internal heap.
Unrelated changes should be removed.
Latest code should be fetched from remote.
Necessary changes should be re-applied to latest code.
Push new commit with a description of what changes are made any why/how they solve the issue.

The compiler message in the issue seems to be providing a hint on its own: -
`error: '_sint_heap_start' undeclared here (not in a function); did you mean '__sint_heap_start'? `

Please add [DRAFT]/[IN PROGRESS] label at the start of the PR title to indicate that changes are yet to be made for this PR.

File: os/arch/arm/src/common/up_internal.h (hunk 6 above)
Comment (sunghan-chang) [reply os/arch/arm/src/common/up_internal.h:282]: @thapav The prefix `_s` means `start`, `_e` does `end`. Then can we remove post fix `_start`?
I think `_s**_heap_start` has two `start`s. Could you check and make a new PR for this if necessary?

File: os/arch/arm/src/stm32l4/stm32l4_ospi.c [hunk 13]
```diff
@@ -1920,9 +1920,7 @@ FAR struct mtd_dev_s *up_flashinitialize(void)
     priv->bwrite = this_bwrite;
     priv->read = this_read;
     priv->ioctl = this_ioctl;
-#ifdef CONFIG_MTD_REGISTRATION
-    priv->mtd.name = "stm32l4_flash";
-#endif
```
Comment (abhishek-samsung) [inline os/arch/arm/src/stm32l4/stm32l4_ospi.c:1925]: We should not just simply remove the line. If you check mtd_dev_s structure you can see that it was changed to ```priv->name``` instead of ```priv->mtd.name```. 

https://github.com/Samsung/TizenRT/blob/436a98058060741956e3b7bc347f96553549bcd3/os/include/tinyara/fs/mtd.h#L168

File: os/tools/check_package_size.py [hunk 14]
```diff
@@ -113,8 +113,9 @@ def check_binary_header(bin_type):
 COMMON_PART_TMP_SIZE = FLASH_SIZE
 APP1_PART_TMP_SIZE = FLASH_SIZE
 APP2_PART_TMP_SIZE = FLASH_SIZE
-
+size=len(SIZE_LIST)
 for name in NAME_LIST :
```
Comment (abhishek-samsung) [inline os/tools/check_package_size.py:117]: This error is because of the kernel not found in the partition list. Please check if the following configs are present. I am attaching another boards defconfig for reference.

https://github.com/Samsung/TizenRT/blob/436a98058060741956e3b7bc347f96553549bcd3/build/configs/rtl8721csm/hello/defconfig#L266-L268

File: build/configs/stm32l4r9ai-disco/pm/defconfig [hunk 15]
```diff
@@ -1250,4 +1375,10 @@ CONFIG_IOTBUS_UART=y
 #
 # Security Framework
 #
-# CONFIG_SECURITY_API is not set
+CONFIG_SECURITY_API=y
```
Comment (abhishek-samsung) [inline build/configs/stm32l4r9ai-disco/pm/defconfig:1378]: Please match the new defconfig with the old one. Like this config was not set in the original defconfig so don't set it in the new config as well.

File: apps/shell/tash_main.c [hunk 16]
```diff
@@ -121,12 +121,12 @@ char *tash_read_input_line(int fd)
 	int nbytes;
 	int char_idx;
 	bool is_tab_pressed = false;
-#if TASH_MAX_STORE > 0
 	bool is_esc_pressed = false;
+	int prev_cmd_len = 0;
+#if TASH_MAX_STORE > 0
```
Comment (abhishek-samsung) [inline apps/shell/tash_main.c:126]: This doesn't seem right. Is ```CONFIG_TASH_MAX_STORE_COMMANDS``` set?
